src/vs/base/test/node/encoding/encoding.test.ts

```diff
@@ -322,11 +319,91 @@ suite('Encoding', () => {
 	test('toDecodeStream - empty file', async function () {
 
 		let path = getPathFromAmdModule(require, './fixtures/empty.txt');
-		let source = fs.createReadStream(path);
+		let source = streamToBufferReadableStream(fs.createReadStream(path));
 		let { detected, stream } = await encoding.toDecodeStream(source, { guessEncoding: false, overwriteEncoding: detected => detected || encoding.UTF8 });
 
 		let expected = await readAndDecodeFromDisk(path, detected.encoding);
 		let actual = await readAllAsString(stream);
 		assert.equal(actual, expected);
 	});
+
+	test('toDecodeStream - decodes buffer entirely', async function () {
```

Comment:
This tests ensures we call `decoder.end()`. Not sure how obvious it is from the test implementation. Any suggestions to improve it are welcome.

Reply:
@gyzerok thanks for adding more tests around encoding 👍